runtime/runtime_test.go

```diff
@@ -4460,139 +4462,251 @@ func TestRuntimeBlock(t *testing.T) {
+
+		runValidCaseWithModulo := func(t *testing.T, ty sema.Type) {
+			// build a big Int from the modulo buffer, with the required `ty` size
+			// big.Int are used as they cover all the tested types including the small ones (UInt8 ..)
+			modulo := new(big.Int).SetBytes(moduloBuffer[:typeToBytes[ty]])
+
+			value, err := executeScript(ty, modulo.String(), readCryptoRandom)
+			require.NoError(t, err)
+			// convert `value` to big Int for comparison
+			valueBig, ok := new(big.Int).SetString(value.String(), 10)
+			require.True(t, ok)
+			// check that modulo > value
+			require.True(t, modulo.Cmp(valueBig) == 1)
+		}
+		testAllTypes(t, runValidCaseWithModulo)
+	})
+
+	// test valid edge cases of the value modulo - test all types
+	t.Run("script with modulo edge cases all types", func(t *testing.T) {
+		// case where modulo is the max value of the type
+		runValidCaseWithMaxModulo := func(t *testing.T, ty sema.Type) {
+
+			// set modulo to the max value of the type: (1 << bitSize) - 1
+			// big.Int are used as they cover all the tested types including the small ones (UInt8 ..)
+			bitSize := typeToBytes[ty] << 3
+			one := big.NewInt(1)
+			modulo := new(big.Int).Lsh(one, uint(bitSize))
+			modulo.Sub(modulo, one)
+
+			value, err := executeScript(ty, modulo.String(), readCryptoRandom)
+			require.NoError(t, err)
+			// convert `value` to big Int for comparison
+			valueBig, ok := new(big.Int).SetString(value.String(), 10)
+			require.True(t, ok)
+			// check that modulo > value
+			require.True(t, modulo.Cmp(valueBig) == 1)
+		}
 
-			require.Equal(t, getFixedSizeUnsignedIntegerForSemaType(ty), value)
+		// case where modulo is 1 and expected value in 0
+		runValidCaseWithOneModulo := func(t *testing.T, ty sema.Type) {
+			// set modulo to 1
+			value, err := executeScript(ty, "1", readCryptoRandom)
+			require.NoError(t, err)
+			// check that value is zero
+			require.True(t, value.String() == "0")
+		}
+
+		t.Run("max modulo", func(t *testing.T) {
+			testAllTypes(t, runValidCaseWithMaxModulo)
 		})
-	}
+		t.Run("one modulo", func(t *testing.T) {
+			testAllTypes(t, runValidCaseWithOneModulo)
+		})
+	})
+
+	// function should error if zero is used as modulo - test all types
+	t.Run("script with zero modulo", func(t *testing.T) {
+		runCaseWithZeroModulo := func(t *testing.T, ty sema.Type) {
+			// set modulo to "0"
+			_, err := executeScript(ty, "0", readCryptoRandom)
+			assertUserError(t, err)
+			require.ErrorContains(t, err, stdlib.ZeroModuloError.Error())
+		}
+		testAllTypes(t, runCaseWithZeroModulo)
+	})
+
+	// sanity statistical test to make sure the random numbers less than modulo
+	// are uniform in [0,modulo-1].
+	// The test requires the original random source (here `crypto/rand`)
+	// to be uniform.
+	// The test uses the same small values for all types: one is a power of 2
+	// and the other is not.
+	t.Run("basic uniformity with modulo", func(t *testing.T) {
+		// skipped in CI because the test is slow
+		t.Skip()
```

Comment:
Not sure about this `Skip`. This is a statistical test that runs the script about 80k times to test uniformity and validate that the modulo computation isn't biased. It takes about 20s when run locally, which is significantly slower than the rest of the tests.

It should be still run when the code changes, at least locally.